File: BaseLibrary/command/equipment/GetMotorsCommand.cs
﻿using System.Collections.Generic;

namespace BaseLibrary.command.equipment {
    public class GetMotorsCommand : AEquipmentCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        public GetMotorsCommand() : base() { }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/equipment/GetMotorsCommand.java

package BaseLibrary.communication.command.equipment;

import BaseLibrary.communication.protocol.AProtocol;

public class GetMotorsCommand extends AEquipmentCommand {

	private static final String NAME = "MOTORS";

	static {
		AProtocol.registerForDeserialize(NAME, GetMotorsCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	
	public GetMotorsCommand() {
	}
}
